cmd/buildah/build.go

```diff
@@ -111,6 +111,7 @@ func buildCmd(c *cobra.Command, inputArgs []string, iopts buildOptions) error {
 	if err := auth.CheckAuthFile(iopts.BudResults.Authfile); err != nil {
 		return err
 	}
+	iopts.BudResults.Authfile = buildahutil.MirrorToTempFileIfPathIsDescriptor(iopts.BudResults.Authfile)
```

Comment:
Please do a `defer remove(authfiile)` here to make sure that created tmpfiles get cleaned up in all return paths (including error paths).

Reply:
@vrothberg instead of defer do you think this makes sense 

```go
iopts.BudResults.Authfile, shouldBeCleaned = buildahutil.MirrorToTempFileIfPathIsDescriptor(iopts.BudResults.Authfile)

if shouldBeCleaned {
            buildahutil.RemoveTempFileByPath(iopts.BudResults.Authfile)
   }
   ```

Reply:
I think defer makes sense nvm.

Reply:
I think it should look as follows:
```go
iopts.BudResults.Authfile, shouldBeCleaned = buildahutil.MirrorToTempFileIfPathIsDescriptor(iopts.BudResults.Authfile)
if shouldBeCleaned {
    defer os.Remove(iopts.BudResults.Authfile)
  }
```

* Use `os.Remove()` instead of wrapping that into a custom function.
* We need defer to remove the file in all return paths.